platform/pulpcore/download/single.py

```diff
@@ -101,6 +102,25 @@ def register(self, event, handler):
         """
         self.router.register(event, handler)
 
+    def clone(self):
```

Comment:
Did you consider implementing this as ``__deepcopy__``?

See the last paragraph here for details: https://docs.python.org/3/library/copy.html#copy.deepcopy

Reply:
Yes, but I think `clone()` has the following advantages:
* seems more explicit
* subclasses not required to impl anything.
* one less import for *user*.

Thoughts?

Reply:
+1 That makes sense, especially the part about making it work for subclasses.